smart_contract/scripts/utils/Faucet.ts

```diff
@@ -0,0 +1,18 @@
+import { Signer } from 'ethers'
+import { addressBook, ethers, network } from 'hardhat'
+
+export const deploy = async (deployer: Signer, logAddress: boolean = true) => {
+  const Faucet = await ethers.getContractFactory('Faucet')
+  const faucet = await Faucet.deploy()
+
+  await faucet.waitForDeployment()
+  const faucetAddress = await faucet.getAddress()
+
+  if (logAddress) {
+    await addressBook.saveContract('Faucet', faucetAddress, network.name, await deployer.getAddress())
```

Comment:
saveContract is not a promise so doesn't need to be awaited. Also, where is this saving too? I'm nto finding and address book.
```suggestion
    addressBook.saveContract('Faucet', faucetAddress, network.name, await deployer.getAddress())
```

Reply:
it's now in contractsAddressDeployed.json